dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -158,37 +159,70 @@
         // Get all the tools based on the found namespaces
         List<Tool> apiTools = getToolsFromNamespace(namespaces);
 
+        String registryString = getRegistry().getDockerPath();
+
         // Add manual tools to list of api tools
         User user = userDAO.findById(userId);
-        List<Tool> manualTools = toolDAO.findByMode(ToolMode.MANUAL_IMAGE_PATH);
+        List<Tool> manualTools = toolDAO.findByModeRegistryNamespace(ToolMode.MANUAL_IMAGE_PATH, registryString, organization);
+        List<Tool> notManualTools = toolDAO.findByNotModeRegistryNamespace(ToolMode.MANUAL_IMAGE_PATH, registryString, organization);
```

Comment:
I wonder if it's quicker to just do one query instead of two, then separate the results by ToolMode? Filtering and sorting are a lot faster in the DB, but here, you end up fetching all tool modes anyway, i.e., not filtering on ToolMode.

That said, probably not worth worrying about, and maybe it is faster the current way.